### PR TITLE
ARROW-18159: [Go][Release] Add `go install` to verify-release script

### PR DESCRIPTION
**Pull request by zeroshade**

### CODE REVIEW

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -859,11 +859,12 @@ test_go() {
...
 
-  pushd go/arrow
+  pushd go
   go get -v ./...
-  go test ./...
+  go test ./...  
+  go install ./...  
   go clean -modcache
-  popd
+  popd
```
Comment (cyb70289) [inline dev/release/verify-release-candidate.sh:867]: Looks there's a trailing space.

Do we want a `shellcheck` lint job in CI? No?

**Review by kou — COMMENTED** [review]
Thanks. But this installs `arrow-flight-integration-server` to `~/go/`. Can we install it into `$ARROW_TMPDIR/`?

```diff
diff --git a/dev/release/verify-release-candidate.sh b/dev/release/verify-release-candidate.sh
index 902e4ec713..04f97611f1 100755
--- a/dev/release/verify-release-candidate.sh
+++ b/dev/release/verify-release-candidate.sh
@@ -395,8 +395,13 @@ install_go() {
     return 0
   fi
 
+  local prefix=${ARROW_TMPDIR}/go
+  mkdir -p $prefix
+
   if command -v go > /dev/null; then
     show_info "Found $(go version) at $(command -v go)"
+    export GOPATH=${prefix}/gopath
+    export PATH=$GOPATH/bin:$PATH
     return 0
   fi
 
@@ -416,13 +421,12 @@ install_go() {
     local os=linux
   fi
 
+  pushd $prefix
   local archive="go${version}.${os}-${arch}.tar.gz"
   curl -sLO https://dl.google.com/go/$archive
-
-  local prefix=${ARROW_TMPDIR}/go
-  mkdir -p $prefix
-  tar -xzf $archive -C $prefix
+  tar -xzf $archive -C .
   rm -f $archive
+  popd
 
   export GOROOT=${prefix}/go
   export GOPATH=${prefix}/gopath
```

**Comment by zeroshade** [comment]
Hmm, you said in your case `go` was already installed on the system right? If `go` already exists on the system, at line 398 it looks like it checks for the existence of the `go` command and if it exists, it skips trying to install Go and returns without setting the `GOPATH`/`GOROOT` env vars. I'll make a change to that logic to set the vars and let's see if that works.

**Comment by zeroshade** [comment]
@kou can you try again and see if it properly installs to `$ARROW_TMPDIR`?

**Review by kou — APPROVED** [review]
+1

> you said in your case `go` was already installed on the system right?

Right. I have `go` installed by `apt`.

> can you try again and see if it properly installs to `$ARROW_TMPDIR`?

Sure. I confirmed that `$ARROW_TMPDIR` is used. Thanks!

**Pull request merged**